app/services/induction/delete_record.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+class Induction::DeleteRecord < BaseService
+  class MiddleOfHistoryError < StandardError; end
+  class TransferFlagChangedError < StandardError; end
+  class TrainingStatusChangedError < StandardError; end
+
+  def call
+    ActiveRecord::Base.transaction do
+      check_record_is_deletable!
+
+      update_previous_record
+      delete_induction_record
+    end
+  end
+
+private
+
+  attr_reader :induction_record
+
+  def initialize(induction_record:)
+    @induction_record = induction_record
+  end
```

Comment:
As discussed earlier it seems like entering at the participant profile level makes more sense to me, because `Induction::DeleteRecord` makes it sound like it's simply deleting this record but actually it's doing that **and** potentially amending the onnes around it.

Reply:
Should the purpose of the class be to 'fix' a _participant's_ induction record by manipulating their `InductionRecord` rows? Rather than going induction record by induction record and merging them together.

Reply:
@peteryates I think the intention was that something like you describe would be the tier above this. This service is like removing a record from a linked list (with a few extra checks thrown in) that could be reused in cases where some precision surgery was needed and would generally do the right thing.